osc/osc.go

```diff
@@ -959,7 +959,7 @@ func readPaddedString(reader *bufio.Reader) (string, int, error) {
 	str = str[:len(str)-1]
```

Comment:
This line is the source of the issues.

Basically, OSC uses null terminated strings, which means that the delimiter is part of the calculation here.
> **OSC-string**
>  A sequence of non-null ASCII characters followed by a null, followed by 0-3 additional null characters to make the total number of bits a multiple of 32.

This means that a 4-byte string would actually be 5 bytes, in which case it's padded by an additional 3 bytes. Whereas a 3-byte string would only by 4 bytes, hence not padded at all.

Basically: Line #971 should become `return str[len(str)-1], n, nil`.

See my comments on the tests that you added.

Reply:
Thanks @chabad360 ! I was getting close but unfortunately the OSC spec examples page seems to be gone forever! :cry: